types/knockout/index.d.ts

```diff
@@ -95,6 +95,8 @@ interface KnockoutObservableArrayStatic {
 }
 
 interface KnockoutObservableArray<T> extends KnockoutObservable<T[]>, KnockoutObservableArrayFunctions<T> {
+    <T>(): T[];
```

Comment:
same here.

Reply:
Not sure what you want, have you tried an example from there https://github.com/DefinitelyTyped/DefinitelyTyped/pull/21569#issuecomment-344900695? The same with `KnockoutObservableArray`.

```javascript
/** @type {KnockoutObservableArray<Date>} */
var a = ko.observableArray();
```
1. `a.` should provide proposals for `KnockoutObservableArray<Date>`
2. `a().` - proposals for `Array<Date>`
3. `a()[0].` - proposals for `Date`

Case 1 works, cases 2 and 3 does not work without these changes.

This can be due to some bug in IntelliJ IDEA. But currently this is only solution I found.